service/lib/agama/storage/zfcp/manager.rb

```diff
@@ -54,14 +55,20 @@ def on_disks_change(&block)
 
         # Probes zFCP
         #
-        # Callbacks are called at the end, see {#on_probe}.
+        # Callbacks {#on_probe} are called after probing, and callbacks {#on_disks_change} are
+        # called if there is any change in the disks.
         def probe
           logger.info "Probing zFCP"
 
+          previous_disks = disks
           yast_zfcp.probe_controllers
           yast_zfcp.probe_disks
 
           @on_probe_callbacks&.each { |c| c.call(controllers, disks) }
+
+          return unless disks_changed?(previous_disks)
+
+          @on_disks_change_callbacks&.each { |c| c.call(disks) }
```

Comment:
np: Is it possible that `@on_disks_change_callbacks` is nil? Not something to change in this PR, but I rather prefer to have an empty array.

Reply:
Yes, because is initialize on assignement, see https://github.com/openSUSE/agama/blob/master/service/lib/agama/storage/zfcp/manager.rb#L51. We could avoid it by initializing the list in the constructor or by adding a `#on_disk_change_callbacks` method.